pkg/controller/networkpolicy/validate_test.go

```diff
@@ -1528,14 +1616,236 @@ func TestValidateAntreaGroup(t *testing.T) {
+			},
+			oldGroup: &crdv1alpha3.Group{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "anp-group-set-with-podselector-and-ipblock",
+					Namespace: "x",
+				},
+				Spec: crdv1alpha3.GroupSpec{
+					PodSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"foo=": "bar"},
+					},
+					IPBlocks: []crdv1alpha1.IPBlock{
+						{CIDR: "10.0.0.10/32"},
+					},
+				},
+			},
+			operation: admv1.Update,
+		},
+		{
+			name: "anp-group-to-delete",
+			oldGroup: &crdv1alpha3.Group{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "anp-group-set-with-podselector-specified",
+					Namespace: "x",
+				},
+				Spec: crdv1alpha3.GroupSpec{
+					PodSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"foo=": "bar"},
+					},
+				},
+			},
+			operation: admv1.Delete,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, controller := newController()
+			validator := NewNetworkPolicyValidator(controller.NetworkPolicyController)
+			actualReason, allowed := validator.validateAntreaGroup(tt.curGroup, tt.oldGroup, tt.operation, authenticationv1.UserInfo{})
+			assert.Equal(t, tt.expectedReason, actualReason)
+			if tt.expectedReason == "" {
+				assert.True(t, allowed)
+			} else {
+				assert.False(t, allowed)
+			}
+		})
+	}
+}
+
+func TestValidateTier(t *testing.T) {
+	tests := []struct {
+		name           string
+		curTier        *crdv1alpha1.Tier
+		oldTier        *crdv1alpha1.Tier
+		operation      admv1.Operation
+		expectedReason string
+	}{
+		{
+			name: "create-tier-pass",
+			curTier: &crdv1alpha1.Tier{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tier-priority-3",
+					Namespace: "x",
+				},
+				Spec: crdv1alpha1.TierSpec{
+					Priority: 3,
+				},
+			},
+			operation: admv1.Create,
+		},
+		{
+			name: "create-tier-failed-with-reserved-priority",
+			curTier: &crdv1alpha1.Tier{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tier-priority-251",
+					Namespace: "x",
+				},
+				Spec: crdv1alpha1.TierSpec{
+					Priority: 251,
+				},
+			},
+			operation:      admv1.Create,
+			expectedReason: "tier tier-priority-251 priority 251 is reserved",
+		},
+		{
+			name: "delete-tier-pass",
+			oldTier: &crdv1alpha1.Tier{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tier-priority-3",
+					Namespace: "x",
+				},
+				Spec: crdv1alpha1.TierSpec{
+					Priority: 3,
+				},
+			},
+			operation: admv1.Delete,
+		},
 	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, controller := newController()
+			validator := NewNetworkPolicyValidator(controller.NetworkPolicyController)
+			actualReason, allowed := validator.validateTier(tt.curTier, tt.oldTier, tt.operation, authenticationv1.UserInfo{})
+			assert.Equal(t, tt.expectedReason, actualReason)
+			if tt.expectedReason == "" {
+				assert.True(t, allowed)
+			} else {
+				assert.False(t, allowed)
+			}
+		})
+	}
+}
+
+func TestValidateMulticastIGMP(t *testing.T) {
```

Comment:
why creating a separate test just for validateMulticastIGMP? It looks not different from the other validation. Perhaps just add the cases to TestValidateAntreaPolicy.

Reply:
@jainpulkit22 Could you check and address a few old comments from @tnqn, I didn't see they are being addressed.
You may check https://github.com/antrea-io/antrea/pull/4629/files to get the leftover comments.